### ISSUE TITLE
solve(Eq(sin(a*x),cos(x)),x)

### ISSUE THREAD

**Issue by WoosterUK** [sympy/sympy#11739]
Running the above command: in Python 3.4, SymPy produces the correct answers, `[-I*log(-(1 - I)/(1 + I))/(a - 1), -I*log(-(1 - I)/(1 + I))/(a + 1)]`, along with two incorrect ones, `[pi, pi/a]`; in Python 2.7, this works fine and does not produce the erroneous answers.

**Comment by ArifAhmed1995** [comment]
Well for me , both Python 2 and 3 return the extra incorrect answers : 
```
arif7@arif7-PC ~ $ python --version
Python 2.7.11+
arif7@arif7-PC ~ $ python3 --version
Python 3.5.1+
arif7@arif7-PC ~ $ cd sympy/
arif7@arif7-PC ~/sympy $ python test.py
[pi, pi/a, -I*log(-(1 - I)/(1 + I))/(a - 1), -I*log(-(1 - I)/(1 + I))/(a + 1)]
arif7@arif7-PC ~/sympy $ python3 test.py
[pi, pi/a, -I*log(-(1 - I)/(1 + I))/(a - 1), -I*log(-(1 - I)/(1 + I))/(a + 1)]

```
Using a debugger , I guess the problem lies in `f = factor(powdenest(lhs - rhs))` line of the `_tsolve` function in solvers.py. After this line executes factors : `(1 + exp(I*x))` and `(1 + exp(I*a*x))` appear along with others containing correct solutions.

I'll come back to this issue soon after my mid-semester exams get over.

**Comment by ArifAhmed1995** [comment]
@WoosterUK By the way, how did you installed sympy ? I reckon you didn't a global install for both Python 2 and 3, otherwise the issue would rise up for both versions.

**Comment by WoosterUK** [comment]
No, I installed python2, then sympy (using pip, I think), then python3, pip3 and then sympy for python3. In python2, I was using sympy 0.7.6.1. I have upgraded to sympy 1.0, and the bug now appears! So not a python3 issue, but a sympy 1.0 issue.

**Comment by oscarbenjamin** [comment]
Current master gives
```julia
In [138]: solve(Eq(sin(a*x),cos(x)),x)                                                                                                                        
Out[138]: 
⎡       π          π      π⎤
⎢π, ─────────, ─────────, ─⎥
⎣   2⋅(a - 1)  2⋅(a + 1)  a⎦
```
so now the correct answers are gone and there are 4 incorrect answers instead of 2.
